ir/src/trace_columns.rs

```diff
@@ -0,0 +1,36 @@
+use super::SemanticError;
+use std::collections::BTreeMap;
+
+/// A representation of the trace columns declared for the AIR.
+#[derive(Debug, Default)]
+pub struct TraceColumns {
+    /// A map of a set of trace columns using the declared identifier as the key and the column
+    /// index as the value.
+    columns: BTreeMap<String, usize>,
+}
+
+impl TraceColumns {
+    /// Add a new column by its name and index
+    pub(super) fn insert(&mut self, name: &str, index: usize) -> Result<(), SemanticError> {
+        let result = self.columns.insert(name.to_string(), index);
+        if result.is_some() {
+            Err(SemanticError::DuplicateTraceColumn(format!(
+                "{} was defined more than once",
+                name
+            )))
+        } else {
+            Ok(())
+        }
+    }
+
+    pub(crate) fn get_column_index(&self, name: &str) -> Result<usize, SemanticError> {
```

Comment:
Minor nit: Might be useful to have a doc comment to keep things consistent.

Reply:
Fixed!